staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -764,3 +808,97 @@ func (s *Scheme) addObservedVersion(version schema.GroupVersion) {
 
 	s.observedVersions = append(s.observedVersions, version)
 }
+
+// isListType returns true if the provided Object has a slice called Items
+func isListType(obj Object) bool {
+	// if we're a runtime.Unstructured, check whether this is a list.
+	// TODO: refactor GetItemsPtr to use an interface that returns []runtime.Object
+	if unstructured, ok := obj.(Unstructured); ok {
+		return unstructured.IsList()
+	}
+
+	_, err := getItemsPtr(obj)
+	return err == nil
+}
+
+// GetItemsPtr returns a pointer to the list object's Items member.
+// If 'list' doesn't have an Items member, it's not really a list type
+// and an error will be returned.
+// This function will either return a pointer to a slice, or an error, but not both.
+func getItemsPtr(list Object) (interface{}, error) {
```

Comment:
this duplicated code here concerns me. Something is wrong with the layering.

Why not move `meta.GetItemsPtr` here?

Reply:
Same for EachListItem.

Reply:
> this duplicated code here concerns me. Something is wrong with the layering.
> 
> Why not move meta.GetItemsPtr here?

From the description of a WIP pull: "I'd probably reconcile the pkg/api/meta cycle by pushing these into the pkg/runtime/scheme package.".  I figure I'll expose them and meta will keep it's wrapper to here since that's where they make more sense.